.github/workflows/lint.yml

```diff
@@ -0,0 +1,40 @@
+name: Lint
+on: 
+  pull_request:
+  push:
+    branches:
+      - "main"
+
+jobs:
+  markdown:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: DavidAnson/markdownlint-cli2-action@v11
+        with:
+          globs: '**/*.md'
+
+  rust:
```

Comment:
I don't know how I feel about pulling this out of the rust-specific workflow. I think it should stay in `rust.yml`. Same with `taplo` being mainly used to enforce rust/crate specific config formatting.